### PR TITLE
Exit with 1 on GitLab or GCasC exceptions

### LINKED ISSUE #25: Return a non-zero return code on error
Currently GCasC always seems to return 0, even in case of errors. This is not viable for CI systems as we can't know when issues arise.

**Current behaviour:**

Example is passing invalid credentials

```
gcasc

 _____  _____              _____ 
|  __ \/  __ \            /  __ \
| |  \/| /  \/  __ _  ___ | /  \/
| | __ | |     / _` |/ __|| |    
| |_\ \| \__/\| (_| |\__ \| \__/\
 \____/ \____/ \__,_||___/ \____/
    
[INFO] [gcasc] TEST MODE ENABLED. NO CHANGES WILL BE APPLIED
[INFO] [gcasc] Initializing GitLab client
[INFO] [gcasc] Trying to initialize GitLab client...
[INFO] [gcasc] GitLab version: unknown, revision: unknown
[INFO] [settings] Configuring ApplicationSettings
[ERROR] [settings] Authentication to GitLab failed
HTTP 401, 401 Unauthorized
[INFO] [appearance] Configuring Appearance
[ERROR] [appearance] Authentication to GitLab failed
HTTP 401, 401 Unauthorized
[INFO] [Features] Configuring GitLab Features
[INFO] [Features] Configured: junit_pipeline_view => True
[INFO] [Features] Configured: ci_enable_live_trace => True
[INFO] [Features] Configured: multiple_merge_request_assignees => True
[INFO] [Features] Configured: coverage_report_view => True
[INFO] [Features] Configured: service_desk_custom_address => True
[ERROR] [features] Authentication to GitLab failed
HTTP 401, 401 Unauthorized
/workspace # echo $?
0
```

**Expected behaviour:**

```
gcasc

 _____  _____              _____ 
|  __ \/  __ \            /  __ \
| |  \/| /  \/  __ _  ___ | /  \/
| | __ | |     / _` |/ __|| |    
| |_\ \| \__/\| (_| |\__ \| \__/\
 \____/ \____/ \__,_||___/ \____/
    
[INFO] [gcasc] TEST MODE ENABLED. NO CHANGES WILL BE APPLIED
[INFO] [gcasc] Initializing GitLab client
[INFO] [gcasc] Trying to initialize GitLab client...
[INFO] [gcasc] GitLab version: unknown, revision: unknown
[INFO] [settings] Configuring ApplicationSettings
[ERROR] [settings] Authentication to GitLab failed
HTTP 401, 401 Unauthorized
[INFO] [appearance] Configuring Appearance
[ERROR] [appearance] Authentication to GitLab failed
HTTP 401, 401 Unauthorized
[INFO] [Features] Configuring GitLab Features
[INFO] [Features] Configured: junit_pipeline_view => True
[INFO] [Features] Configured: ci_enable_live_trace => True
[INFO] [Features] Configured: multiple_merge_request_assignees => True
[INFO] [Features] Configured: coverage_report_view => True
[INFO] [Features] Configured: service_desk_custom_address => True
[ERROR] [features] Authentication to GitLab failed
HTTP 401, 401 Unauthorized
/workspace # echo $?
127
```

### PR DESCRIPTION
**Pull request by kravvcu**
Exit immediately with a nonzero code on any error.

Closes #25

### CODE REVIEW

**Comment by kravvcu** [comment]
I've no idea if this is the right way but I think it is. Any error -> immediately fail, right? :|

File: gcasc/__init__.py [hunk 1]
```diff
@@ -83,8 +84,10 @@ def configure(self, target=None):
                     self._configure(configurer)
                 except GcascException as exc:
                     exceptions.handle_gcasc_exception(exc, logger)
+                    sys.exit(1)
```
Comment (filipowm) [inline gcasc/__init__.py:87]: Could you please include that in tests? ;) 

Unfortunately there are no tests for verifying exception handling yet. But imo they should be here: https://github.com/Roche/gitlab-configuration-as-code/blob/master/tests/gcasc_test.py

Comment (kravvcu) [reply gcasc/__init__.py:87]: Done! xD finally

**Comment by filipowm** [comment]
lgtm

**Pull request merged**